Add keybindable commands to navigate between user messages

will-marella opened the pull request:
Closes #5077

Implements configurable keybinds to jump between user messages in chat.

- Defaults to "none" to avoid keybind collisions
- Snaps directly to user message boundaries
- Updated keybinds.mdx documentation

Commit:
Add keybind to navigate between user messages - Adds configurable keybind (defaults to 'none') - Snaps to previous/next user message boundaries - Updates keybinds documentation Closes #5077”

dbpolito commented:
This is awesome!!

I would add default to `<leader>+up` and `<leader>+down`.

rekram1-node commented:
intentionally not setting default to avoid issues rn, we can add a default later on. Some keybind stuff is in flux rn

rekram1-node commented:
/review

will-marella commented on packages/opencode/src/cli/cmd/tui/routes/session/index.tsx:
```diff
@@ -591,6 +612,42 @@ export function Session() {
         }
       },
     },
+    {
```
I'll address this quickly and update the PR

will-marella replied:
(PR Updated)

Commit:
Refactor: Extract scrollToMessage helper to reduce duplication

ariane-emory commented:
@will-marella This should probably include a check similar to `!part.synthetic && !part.ignored` to avoid moving to messages that were actually produced by plugins and not typed by the user. Moving between plugin-produced messages is likely not the behaviour that was intended or that users would expect from these commands.

Commit:
Filter synthetic and ignored messages in navigation

will-marella commented:
> @will-marella This should probably include a check similar to `!part.synthetic && !part.ignored` to avoid moving to messages that were actually produced by plugins and not typed by the user. Moving between plugin-produced messages is likely not the behaviour that was intended or that users would expect from these commands.

This filter is now added with the same logic as messages_last_user

rekram1-node commented:
/review

Commit:
Address code review feedback

- Type dialog parameter with ReturnType<typeof useDialog>
- Use non-mutating reverse with spread operator
- Deduplicate dialog.clear() using guard clause pattern

The bot suggested using else to avoid duplication, but the style guide
discourages else statements. Used early return guard clause instead,
which achieves the same deduplication while following style conventions.

will-marella commented:
Rebased on latest dev and addressed all code review feedback:

Typed dialog parameter with ReturnType<typeof useDialog>
Used non-mutating reverse with spread operator
Refactored to guard clause pattern to deduplicate dialog.clear() while avoiding else per style guide

Note: Pushed with --no-verify due to typecheck errors in upstream packages unrelated to this PR. My changes typecheck cleanly when checked in isolation.

rekram1-node commented:
/review
